library/std/src/io/error/tests.rs

```diff
@@ -67,3 +67,18 @@ fn test_const() {
     assert!(format!("{:?}", E).contains("\"hello\""));
     assert!(format!("{:?}", E).contains("NotFound"));
 }
+
+#[test]
+fn test_error_packing() {
+    for code in -20i32..20i32 {
+        let e = Error::from_raw_os_error(code);
+        assert_eq!(e.raw_os_error(), Some(code));
+    }
+    assert_eq!(Error::from(ErrorKind::NotFound).kind(), ErrorKind::NotFound);
+    assert_eq!(Error::from(ErrorKind::Uncategorized).kind(), ErrorKind::Uncategorized);
+    assert_eq!(Error::from(ErrorKind::NotFound).kind(), ErrorKind::NotFound);
+    assert_eq!(Error::from(ErrorKind::Uncategorized).kind(), ErrorKind::Uncategorized);
+    let dunno = const_io_error!(ErrorKind::Uncategorized, "dunno");
+    assert_eq!(dunno.kind(), ErrorKind::Uncategorized);
+    assert!(format!("{:?}", dunno).contains("dunno"))
```

Comment:
what is this checking?

Reply:
I believe there isn't really anything in the public API that exposes that message right now -- it's just in the output.

I've added a comment to that effect, and also made all these tests verify check that the ErrorData variant (e.g. the portable enum that the packed stuff decodes into) is what's expected (also helps ensure there's not something giving the right answer for the wrong reason)